src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -106,10 +109,39 @@ public void CompilerDataLogger_Constructor_ShouldThrowException_WhenTelemetryIsE
                 { "CompilerTelemetry.Options", compilerOptions }
             };
 
-            Assert.Throws<InvalidOperationException>(() => new CompilerDataLogger(sarifOutputFilePath: string.Empty, context, fileSystem.Object));
+            Assert.Throws<InvalidOperationException>(() => new CompilerDataLogger(sarifOutputFilePath: string.Empty, Sarif.SarifVersion.Current, context, fileSystem.Object));
         }
 
-        private List<ITelemetry> TestSetup(string sarifLogFilePath, BinaryAnalyzerContext context, out CompilerDataLogger logger)
+        [Fact]
+        public void CompilerDataLogger_Dispose_ShouldReadSarifV1()
+        {
+            string sarifLogPath = Path.Combine(PEBinaryTests.BaselineTestDataDirectory, "Expected", "Native_x86_VS2019_SDL_Enabled_Sarif.v1.0.0.sarif");
+            var fileSystem = new Mock<IFileSystem>();
+            string content = File.ReadAllText(sarifLogPath);
+            byte[] byteArray = Encoding.UTF8.GetBytes(content);
+            var context = new BinaryAnalyzerContext() { TargetUri = new Uri(@"c:\file.dll"), ForceOverwrite = true };
+            var compilerOptions = new PropertiesDictionary
+            {
+                { "CsvOutputPath", @$"C:\temp\{Guid.NewGuid()}.sarif" }
+            };
+
+            context.Policy = new PropertiesDictionary
+            {
+                { "CompilerTelemetry.Options", compilerOptions }
+            };
+
+            fileSystem
+                .Setup(f => f.FileOpenRead(It.IsAny<string>()))
+                .Returns(new MemoryStream(byteArray));
+
+            List<ITelemetry> sendItems = TestSetup(sarifLogPath, context, Sarif.SarifVersion.OneZeroZero, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.Dispose();
+
+            fileSystem.Verify(fileSystem => fileSystem.FileOpenRead(sarifLogPath), Times.Once);
+            sendItems.Count.Should().Be(1);
+        }
```

Comment:
This is the added test.
Used the `Native_x86_VS2019_SDL_Enabled.exe` to generate a SARIF v1 output (in addition to the SARIF V2 used in the baseline tests).
Mocked the fileSystem to return this sarif output and verified that the fileSystem was called exactly one time as expected.
Finally, verified that we generated the single telemetry event as expected when calling `CompilerDataLogger.Disepose()`.